pkg/chrootarchive/archive_unix.go

```diff
@@ -79,6 +80,11 @@ func invokeUnpack(decompressedArchive io.Reader, dest string, options *archive.T
 	w.Close()
 
 	if err := cmd.Wait(); err != nil {
+		// when `xz -d -c -q | docker-untar ...` failed on docker-untar side,
+		// we need to exhaust `xz`'s output, otherwise the `xz` side will be
+		// pending on write pipe forever
+		io.Copy(ioutil.Discard, decompressedArchive)
```

Comment:
@cpuguy83 here it is, reexec of "docker-untar ..." failure has to exhaust all input, otherwise the underlying xz pending forever